com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java

```diff
@@ -67,24 +69,37 @@
 
 	@Test
 	public void testSubscribe() {
+		// try to subscribe to a valid modeluri
+		//
 		when(clientCtx.getSessionId()).thenReturn(UUID.randomUUID().toString());
 		when(clientCtx.pathParam("modeluri")).thenReturn("fancytesturi");
+		when(repository.hasModel("fancytesturi")).thenReturn(true);
 
-		assertFalse(sessionController.isClientSubscribed(clientCtx));
-
-		sessionController.subscribe(clientCtx, clientCtx.pathParam("modeluri"));
-
+		assertTrue(sessionController.subscribe(clientCtx, clientCtx.pathParam("modeluri")));
 		assertTrue(sessionController.isClientSubscribed(clientCtx));
+		verify(clientCtx).send(argThat(jsonNodeThat(containsRegex("(?i)\"type\":\"success\""))));
+
+		// try to subscribe to an invalid modeluri
+		//
+		when(clientCtx2.pathParam("modeluri")).thenReturn("tedioustesturi");
+		assertFalse(sessionController.subscribe(clientCtx2, clientCtx2.pathParam("modeluri")));
+		assertFalse(sessionController.isClientSubscribed(clientCtx2));
 	}
 
 	@Test
 	public void testUnsubscribe() {
 		testSubscribe();
```

Comment:
I'd prefer if we wouldn't call another test directly here, extract method instead.